extra/training.py

```diff
@@ -34,7 +24,7 @@ def train(model, X_train, Y_train, optim, steps, BS=128, lossfn=sparse_categoric
     # printing
     if not noloss:
       cat = np.argmax(out.numpy(), axis=-1)
-      accuracy = (cat == y).mean()
+      accuracy = (cat == y.numpy()).mean()
```

Comment:
Hmm, how come this didn't throw an error?

Reply:
It looks like the problem is on the numpy end, even if I put an Assert inside tensor.py `__eq__`, it won't able to catch it. I can putting an assert in this file, but it seems a bit pointless though. 
```
x = np.array([1,1,1])
x_tiny = Tensor([1,2,3])
print(x == x_tiny)
print((x_tiny == x).numpy())
```
[ True  True  True]
[1. 0. 0.]